client/state/utils.js

```diff
@@ -105,6 +105,11 @@ export function createReducer( initialState = null, customHandlers = {}, schema
 	return ( state = initialState, action ) => {
 		const { type } = action;
 
+		if ( 'production' !== process.env.NODE_ENV && 'type' in action && ! type ) {
+			throw new TypeError( 'Reducer called with undefined type.' +
+				' Verify that you\'ve defined your action type in state/action-types.js' );
```

Comment:
>How about double quotes/backtick here to avoid the need to escape?

I changed the text a bit in the rebased 51ecd72 to avoid the need for escaping.